powdr_cli/src/main.rs

```diff
@@ -476,25 +484,41 @@ fn run_riscv_asm<F: FieldElement>(
         output_dir,
         force_overwrite,
         prove_with,
+        vec![],
     )?;
     Ok(())
 }
 
+#[allow(clippy::too_many_arguments)]
 fn compile_with_csv_export<T: FieldElement>(
     file: String,
     output_directory: String,
+    witness_values: Option<String>,
     inputs: String,
     force: bool,
     prove_with: Option<BackendType>,
     export_csv: bool,
-    csv_mode: CsvRenderMode,
+    csv_mode: CsvRenderModeCLI,
 ) -> Result<(), Vec<String>> {
+    let external_witness_values = witness_values
+        .map(|csv_path| {
+            let csv_file = fs::File::open(csv_path).unwrap();
+            let mut csv_writer = BufReader::new(&csv_file);
+            read_polys_csv_file::<T>(&mut csv_writer)
+        })
+        .unwrap_or(vec![]);
+
+    // Convert Vec<(String, Vec<T>)> to Vec<(&str, Vec<T>)>
```

Comment:
In my new PR I'm changing a lot of these column names as &str to String, so if you want to change more of that stuff feel free. Otherwise I can do it in my PR

Reply:
Ah cool! This would propagate all the way down to witgen, so I'll leave that for another PR. But the `write_polys_csv_file()` function I introduces already receives a `&[(String, Vec<T>)]`, so you won't have to change that!

Reply:
yea that's why I thought of that